Compile for multiple Android versions at the same time

peat-psuwit opened the pull request:
This PR uses the same technique with ubports/pulseaudio-modules-droid#8 and is required to be compatible with that PR.

peat-psuwit committed:
Revert "Build against android-headers-28"

This reverts commit 0667fa3f41dadd761ab7dda632612a016fe6dfe0.

peat-psuwit committed:
[hidl] migrate to macro-based symdef

This makes the module uses the macros in the newer PA's module.h if
available, and the vendored version otherwise. This also makes
dynamically naming module easier.

peat-psuwit committed:
[hidl] Support adding suffix to module name & specify android headers pkg name

This allows distributions to compile & ship multiple versions of the
module in the same rootfs, choosing between them at runtime.

NotKit commented on debian/control:
```diff
@@ -1,21 +1,30 @@
                audiosystem-passthrough-dev
 Standards-Version: 3.8.4
 
+Package: pulseaudio-modules-droid-hidl-24
```
You do not need to build it with android-headers-24. This module relies on binderized HIDL interfaces which became available in Android 8, and while it may compile, it will not work on earlier devices.

peat-psuwit replied:
Ok, will remove building for -24. I thought audiosystem-passthrough also work as AudioFlinger on Android 7 too (given that AudioFlingerGlue is stopped).

peat-psuwit committed:
[debian] compile for multiple Android versions at the same time

This uses the same technique used by qtubuntu to compile both -android
and -desktop variant at the same time.

It currently builds for API version 28 only, as API version 24 does not
need this package. To add more versions, edit the version list at the
top of the debian/rules file, add packages in the debian/control, and
finally link the .install files.

dobey commented:
Is this still relevant? We have a different branch for the android9 stuff now right? We don't use this `xenial_-_edge_-_android8` anymore?

mariogrip commented:
I would still like to get this in, now that we got both android9 and 10 this is needed

peat-psuwit commented:
Due to branch name change, superseded by #3.
